kaitai_utils.py

```diff
@@ -24,6 +25,24 @@ def get_ksy_file():
         raise ValueError("ERROR: Found ", str(len(ksy_file_list)), " .ksy files in /structs, which is not exactly 1.")
     return os.path.join(path, ksy_file_list[0])
 
+def create_bytearray_child_traces(data_binary: BinaryIO, class_type: Type[KaitaiStruct], trace: ExtractionTrace, path: str):
+    parsed_kaitai_struct = class_type.from_io(data_binary)
+    bytearrays_to_traces(parsed_kaitai_struct, trace, path)
+
+
+def bytearrays_to_traces(instance: Any, trace: ExtractionTrace, path: str):
+    parameters_dict = _parameters_dict(instance)
+    for key, value_object in parameters_dict.items():
+        if not key.startswith("_") and value_object is not None:
```

Comment:
can you extract an 'explaining' variable. Something like is_public_property? Maybe you have a better name

Reply:
done